incorporate lyrics in miniplayer

3r3n-n opened the pull request:
This PR addresses issue [#859](https://github.com/nukeop/nuclear/issues/859) to show the current track lyrics inside the miniPlayer.

It involves the following changes:

- Updated the `MiniTrackInfo` component so it takes the `LyricsView` component as a prop in (`packages/ui/lib/components/MiniPlayer/MiniTrackInfo/index.tsx`) and displays it next to the cover art (`packages/ui/lib/components/MiniPlayer/MiniTrackInfo/styles.scss`).
- Since `MiniTrackInfo` already shows the track title and artist in a pretty way and `LyricsView` also shows them in the header, showing title and artist twice in the mini-player would look weird, so `LyricsView` was updated with an extra prop (boolean) so the `LyricsHeader` shows conditionally (`true` for `LyricsContainer` and `false` for `MiniTrackInfo`).
- Updated the `miniPlayer` snapshot for test in `packages/ui/test/__snapshots__/miniPlayer.test.tsx.snap` to take into account this layout change.
- Updated the third test for `LyricsContainer` in `packages/app/app/containers/LyricsContainer/LyricsContainer.test.tsx` so it simulates a case with an existing track with empty lyrics data.

The result of this implementation looks like this:

https://user-images.githubusercontent.com/82424931/124523979-d317a900-ddbe-11eb-914d-07cb55e249f8.mov

nukeop commented:
Thanks, it looks pretty good. What does it look like on smaller widths, does it scale or change the layout?

3r3n-n commented:
Yes, it scales. For example:

https://user-images.githubusercontent.com/82424931/124526260-e713d880-ddc7-11eb-90b4-c2b2d1df25d5.mov

nukeop commented on packages/i18n/src/locales/ar_SA.json:
```diff
@@ -132,8 +138,12 @@
     "rename": "Rename this playlist",
-    "songs": "{{total}} song",
-    "songs_plural": "{{total}} songs",
+    "songs_0": "{{total}} songs",
```
This won't work since we use songs and songs_plural

3r3n-n replied:
Done :) Just erased the `songs_X` lines and restored songs and `songs_plural` (in the 7 locales files where I found this). 

Not sure why that happened, though (I did not intend to modify locales files in this PR). 

PS: I did an extra commit for this because the first one was not passing the CI checks, some other files were missing the `songs` and/or `songs_plural` lines (but had no `songs_X` lines) so I added them there as well and then it passed.

nukeop commented on packages/app/app/containers/MiniPlayerContainer/index.tsx:
```diff
@@ -31,6 +45,14 @@ const MiniPlayerContainer: React.FC = () => {
         style={{
           zIndex: isMiniPlayerEnabled && 1000
         }}
+        lyricsComponent={
```
Since you pass an element, not a component, rename this prop accordingly

3r3n-n replied:
Done :)

nukeop commented:
Could you please add some tests for this?

3r3n-n commented:
> Could you please add some tests for this?

Since there was already a test for the mini-player, I updated its respective snapshot and verified that it passed npm test. Is that ok ?

nukeop commented:
The snapshot should be updated, but there isn't anything in the lyrics box in it, and it doesn't check fetching lyrics, so it doesn't test this functionality.

3r3n-n commented:
Ooh, I see. I wiill write a test for that and learn more about snapshots since I was misunderstanding how the update works. Thanks !

3r3n-n commented:
Hii, @nukeop !
I just added the tests for the functionality in a new file:
`packages/app/app/containers/MiniPlayerContainer/MiniPlayerContainer.test.tsx`

And updated the UI test in `packages/ui/test/miniPlayer.test.tsx` and its respective snapshot so it includes mock lyrics.
All the tests (including these new ones) are passing.

nukeop commented:
Great, it looks like your changes increase coverage. Although you should not merge master back into your pull request, that causes the diff to include the changes that were made to master in the meantime, and it's not clear which were made by you.

3r3n-n commented:
Oops :s 
I did an override of master into my branch (with `-s ours`) because when I merged it normally, the build crashed so I did the override and then restored my changes from my previous commits to re-integrate them
 but seems like it was a bad idea from my part :s 
Do you have a recommendation on how I should fix this ?

nukeop commented:
Usually you can just leave merging conflicts with master to me. You can try reverting that merge commit now.

3r3n-n commented:
> Usually you can just leave merging conflicts with master to me. You can try reverting that merge commit now.

Done.
Reverted to my last commit (`808c3a3`) before my merge/override mess... but at that point it did not include the new tests yet so I just added them in a few newer commits.

nukeop commented:
There are still some changes to the translation files which should be removed altogether or reverted.

3r3n-n commented:
Ok, just fixed the conflicts in the 3 translation files.

nukeop commented on docs/i18n.md:
```diff
@@ -1514,20 +1514,6 @@
  - year
 
 ### app:
```
This diff should not be here

nukeop commented:
There are still many language files in the PR, which should not be there (as far as I see you did not add any new strings).

HaiDang666 commented:
you can `reset --soft` then remove all the language files. Then commit again if it generates language files again, don't commit them.

3r3n-n commented:
Thanks !
I got quite confused while trying to fix this, so sorry about that. 
I tried reverting almost all commits because I realized that the weird non-intended changes to the translation files happened after my very first commit in this branch. 
I thought that first I should be aiming for no translation files in the _'files changed'_ tab in the github web UI (and then I could re-add my changes) but have not been able to achieve this with the reverts or by pulling master’s latest updates (this last thing just increased the _files changed_ number)… at least now there are no weird changes in the translation files like `artist_0`, `artist_1` and so on. So I was wondering if this is ok and I should re-add my changes on top of this ? Just want to make sure.

Sorry about all this, I am very new to git and it is my first time trying to go back in commits that I already pushed. I will keep trying to fix this in the following days, sorry for leaving my last push in a weird incomplete version, really need to sleep now.

nukeop commented:
I think I have write access to your branch so I will try to make sense of it and try to clean it up today. This makes me think that we should add some information on resolving conflicts and committing files so it's not confusing for new people.

nukeop commented:
The scripts that automatically add missing strings to translations can be removed now too, since translations will be handled on Crowdin, and they introduce a lot of confusion.

3r3n-n commented:
Thank you so much !
I’ll be on the lookout in case there is anything that you need me to do to help fix this.

nukeop commented:
I made a parallel PR since I'm going to do some force pushes and I don't want to overwrite your work: https://github.com/nukeop/nuclear/pull/1014

I'll let you know once I figure out how to make sense of it, then you can just reset everything to the state of that branch, and force push to your branch, so it ends up in your PR.

nukeop commented:
Ok @3r3n-n , I think #1014 should be what we want now, there are only 10 files changed. Can you confirm that you're happy with that?

If you are, you can either force push that branch to this PR's branch, or we can merge the other one (not sure if you'll show up as the author if I squash it though).

3r3n-n commented:
Awesome, thank you so so much !!
I will do the force push from 1014 then, I’m checking how to do it.

HaiDang666 commented:
merge master to your brand will fix the type error from CI

nukeop commented:
No, merging master to this branch is what caused the problem in the first place. We want to merge this branch to master, not vice versa. If anything, you can try rebasing (if you know what you're doing).

3r3n-n commented:
Ok, I did the force push. 
I just ran `npm start` and `npm test` locally just in case, everything passed in my local but I am not sure what might be causing the checks here to fail… well, the ones with the _push_ label because the _pull_request_ ones just passed.

HaiDang666 commented:
it failed cos of https://github.com/nukeop/nuclear/commit/f4bc7cc893b9228320fd24d2c3f269c55fc00cf6
